torch/testing/_internal/common_utils.py

```diff
@@ -2567,6 +2567,10 @@ def to_list(input):
         if isinstance(y, torch.Tensor) and y.is_nested:
             y = y.unbind()
 
+        # TODO: explain why this is needed
+        if isinstance(x, torch.TypedStorage) and isinstance(y, torch.TypedStorage):
+            x, y = [[storage._getitem(idx) for idx in storage._size()] for storage in [x, y]]
```

Comment:
Ok, this is not the patch we want. It only works in case we only pass `TypedStorage`'s directly. As soon as we pass it inside a container like a list, this check never triggers. In e9fc4ad310eea7e020bc6888d68a56f71b7e23a7, I've opted to simply convert the storages into tensors for comparison.